src/org/webonise/multithreading/synchronization/Application.java

```diff
@@ -0,0 +1,53 @@
+package org.webonise.multithreading.synchronization;
+
+public class Application {
+
+    public void start(){
+        runThreads();
+    }
+
+    private void runThreads(){
+        runThreadsWithoutSynchronization();
+        runThreadsWithSynchronization();
+    }
+
+    private void runThreadsWithoutSynchronization(){
+        Thread[] threads = new Thread[5];
+        ThreadPrinter threadPrinter = new ThreadPrinter();
+
+        System.out.println("\nResult of access without synchronization :");
+
+        for (int i=0; i<threads.length; i++){
+            threads[i] = new Thread(new WorkerThread(threadPrinter), "Thread " + i);
+            threads[i].start();
+        }
+
+        for (int i=0; i<threads.length; i++){
+            try {
+                threads[i].join();
```

Comment:
No the loop with continue only after the thread execution is complete. I debugged it. It works same as sequential join statements.